### DISCUSSION TITLE
Git Workflow: Branching, Pull Requests, Reviews, and Merging

### DISCUSSION THREAD

**Discussion by hyunnnn98** [2P3S/scrum-dice#17] [Announcements]
##  Git Workflow: Branching, Pull Requests, Reviews, and Merging

![image](https://user-images.githubusercontent.com/52916934/227955188-dba4f0dc-5549-44a5-9ba1-806333076685.png)

### 1. 브랜치 생성 및 작업

- 개발을 시작하기 전에, 새로운 브랜치를 생성하고 해당 브랜치에서 작업합니다. 
   - 이렇게 하면 본인의 작업과 팀원의 작업이 충돌하는 것을 방지할 수 있습니다.

### 2. 풀 리퀘스트(PR) 생성

 - 작업이 완료되면, 해당 브랜치에서 풀 리퀘스트(PR)를 생성합니다.
 - PR 제목과 설명을 작성하여 팀원들이 이해할 수 있도록 합니다.

### 3. 리뷰 요청

- PR이 생성되면, 다른 팀원들에게 리뷰를 요청합니다. 
- 리뷰어를 지정하고, 리뷰어가 PR을 검토하고 댓글을 작성할 수 있도록 합니다.

### 4. 리뷰 반영

 - 리뷰어들이 작성한 댓글을 확인하고, 필요한 수정을 수행합니다.
 - PR에 대한 댓글에 답변을 작성하여, 다른 팀원들과 의사소통할 수 있도록 합니다.

### 5. 코드 승인

 - 리뷰어들이 모두 승인한 후, PR을 코드 저장소에 병합(merge)합니다.
 - 머지 커밋에 대한 설명을 작성하고, 코드 저장소의 기능을 유지하면서, 병합하는 것이 중요합니다.

### 6. 브랜치 삭제

 - 병합이 완료되면, 해당 브랜치를 삭제합니다. 
    - 이렇게 하면 코드 저장소가 깔끔하게 유지됩니다.

**Comment by Samsam-lee** [comment]
Git Flow
---
![git_flow](https://user-images.githubusercontent.com/60730831/228178211-6dbcb227-4577-4fa2-b9a1-e0589815c7ee.png)

Github Flow
---
![github_flow](https://user-images.githubusercontent.com/60730831/228178257-939c754e-18e5-44dc-85ee-8d28ae08ce6a.png)

Gitlab Flow
---
![gitlab_flow](https://user-images.githubusercontent.com/60730831/228178317-9d1d3ec5-9b20-44d3-af05-17858eca7a06.png)

> 승현이가 말하는 건 Github Flow 인 것 같네!
> Github Flow는 관리로 인해 CI/CD가 필수
> CI/CD도 넣을거구 브랜치도 크게 나뉘는게 없어서 제일 관리하기 좋을 것 같음!!!
